packages/next-auth/src/providers/tiktok.ts

```diff
@@ -0,0 +1,107 @@
+import { TokenSet } from "src"
+import type { OAuthConfig, OAuthUserConfig } from "."
+
+/** @see https://developers.tiktok.com/doc/tiktok-api-v2-get-user-info/ */
+export interface TiktokProfile extends Record<string, any> {
+  data?: Data
+  error?: Error
+}
+
+export interface Data {
+  user?: User
+}
+
+export interface User {
+  display_name?: string
+  following_count?: number
+  is_verified?: boolean
+  open_id?: string
+  union_id?: string
+  video_count?: number
+  avatar_url?: string
+  bio_description?: string
+  follower_count?: number
+  likes_count?: number
+  profile_deep_link?: string
+  username?: string
+}
+
+export interface Error {
+  code?: string
+  message?: string
+  log_id?: string
+}
+
+export default function Tiktok<P extends TiktokProfile>(
+  options: OAuthUserConfig<P>
+): OAuthConfig<P> {
+  return {
+    id: "tiktok",
+    name: "TikTok",
+    type: "oauth",
+    authorization: {
+      url: "https://www.tiktok.com/v2/auth/authorize",
+      params: {
+        client_key: options.clientId,
+        scope: "user.info.profile",
+        response_type: "code",
+      },
+    },
+
+    token: {
+      async request({ checks, client, params, provider }) {
+        const res = await fetch(`https://open.tiktokapis.com/v2/oauth/token/`, {
+          method: "POST",
+          headers: {
+            "Cache-Control": "no-cache",
+            "Content-Type": "application/x-www-form-urlencoded",
+          },
+          body: new URLSearchParams({
+            client_key: provider.clientId!,
+            client_secret: provider.clientSecret!,
+            code: params.code!,
+            grant_type: "authorization_code",
+            redirect_uri: provider.callbackUrl!,
+          }),
+        }).then((res) => res.json())
+
+        const tokens: TokenSet = {
+          access_token: res.access_token,
+          expires_at: res.expires_in,
+          refresh_token: res.refresh_token,
+          scope: res.scope,
+          id_token: res.open_id,
+          token_type: res.token_type,
+          session_state: res.open_id,
+        }
+        return {
+          tokens,
+        }
+      },
```

Comment:
seems like we can set the `client` option and just set an `url` for `token`, unless I'm missing something.
```suggestion
    token: "https://open.tiktokapis.com/v2/oauth/token/",
   client: {
       token_endpoint_auth_method: "client_secret_post",
   }
```

Reply:
the tricky part here is with the "client_secret_post" it will look for a client_id and a client_secret and add those name and values in the body of the request, but tiktok uses client_key and client_secret